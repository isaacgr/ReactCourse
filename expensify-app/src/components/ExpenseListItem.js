import React from 'react';
import {Link} from 'react-router-dom';

const ExpenseListItem = ({ dispatch, id, description, amount, createdAt }) => (
  <div>
    <div>
      <Link
        to={`/edit/${id}`}
        >
        <h3>{description}</h3>
        </Link>
      <p>Amount: {amount}</p>
      <p>Created At: {createdAt}</p>
    </div>
  </div>
);

export default ExpenseListItem;
