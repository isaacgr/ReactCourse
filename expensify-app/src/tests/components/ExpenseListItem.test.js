import React from 'react';
import { shallow } from 'enzyme';
import expenses from '../fixtures/expenses';
import ExpenseListItem from '../../components/ExpenseListItem';

test('should render an expense item', () => {
  const wrapper = shallow(
    <ExpenseListItem
      key={expenses[0].id}
      {...expenses[1]}
  />)
  expect(wrapper).toMatchSnapshot();
})
