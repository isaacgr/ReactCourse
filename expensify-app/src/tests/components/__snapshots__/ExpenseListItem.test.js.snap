// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render an expense item 1`] = `
<div>
  <div>
    <Link
      replace={false}
      to="/edit/2"
    >
      <h3>
        Rent
      </h3>
    </Link>
    <p>
      Amount: 
      10000
    </p>
    <p>
      Created At: 
      -345600000
    </p>
  </div>
</div>
`;
